### PR TITLE
SSO StoryBook WIP

### PR DESCRIPTION
**Pull request by duttonw**
Show sso more visibly per https://servicesmadesimpler.govnet.qld.gov.au/wiki/display/QOL/Form.io+OICD+IDB+integration+POC+with+Freight+Pass

### CODE REVIEW

**Comment by duttonw** [comment]
@devonpis  still not complete as i've not work out how to make the 'login' swap form to 'login' form work correctly (i think it might be doing a refresh of some sort and bugging out storybook)

File: src/components/SingleSignOn/SingleSignOn.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      console.log(`Loaded form: ${form.formio.formId}`);
+      //console.log(JSON.stringify(form.formio));
+      form.on("submitDone", (submission) => {
+
+        Formio.currentUser().then(user => {
+          //clean up URL paramters from submission or logout redirect
+          var formioCleanURL = window.location.href.split("?")[0];
+
+          //trigger post-login form to load
+          window.location.href = formioCleanURL;
```
Comment (devonpis) [inline src/components/SingleSignOn/SingleSignOn.js:68]: storybook is using iframe, so window.location may not work (unless the iframe opened in a new tab)

**Review by devonpis — COMMENTED** [review]
i think it's better to create another folder for this usage example rather than put it in `components`
ie. `src/examples/SingleSignOn/SingleSignOn.js`

**Review by devonpis — APPROVED** [review]
there's still minor lint issues, but will look at it in later PRs

File: src/examples/SingleSignOn/SingleSignOn.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { Canvas, Meta, Story } from "@storybook/addon-docs";
+import { createForm, createBuilder } from "../../utils";
+import SSOHighLevel from "./../../stories/assets/SSO-HighLevel.png";
+import SSOUserFlow from "./../../stories/assets/SSO-UserFlow.png";
+import { createSSoForm } from "./SingleSignOn";
+
+<Meta title="Components/SingleSignOn" />
```
Comment (devonpis) [inline src/examples/SingleSignOn/SingleSignOn.stories.mdx:7]:
```suggestion
<Meta title="Examples/SingleSignOn" />
```

**Comment by devonpis** [comment]
if you try http://localhost:6006/iframe.html?id=components-singlesignon--demo&viewMode=story
to open the story in a window tab istead of an iframe
and click `logout` does it have the expected result?
as i've tried it and doesn't seem to log me out

**Comment by devonpis** [comment]
for the display problem in storybook iframe
i think it's due to the page going to embed in iframe (logout page in this case) has `X-Frame-Options` HTTP response header set to `sameorigin`
https://medium.com/@alexcambose/bypassing-x-frame-options-4934dd852618

**Comment by duttonw** [comment]
#19  completed this work.

**Pull request closed without merging**